### ISSUE TITLE
Apparent Timestep-Dependent Plant Volume Autocalculation

### ISSUE THREAD

**Issue by Myoldmopar** [NREL/EnergyPlus#4188]
An email from @larryscheier described a possible issue with plant volume autocalculation.

I was doing some CPU timings for Trane using the RefBldgHospital example and noticed this: 

| Plant Loop Name | Number of Timesteps: | 1 | 2 | 3 | 4 | 6 | 10 | 15 | 20 | 30 | 60 |
| --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- |
| COOLSYS1 | Plant Loop Volume [m3] | 420.7 | 209 | 139.2 | 104.4 | 69.8 | 41.8 | 27.9 | 20.9 | 14 | 7 |
| HEATSYS1 | Plant Loop Volume [m3] | 129.2 | 65.7 | 43.9 | 33 | 21.9 | 13.2 | 8.8 | 6.6 | 4.4 | 2.2 |
| SWHSYS1 | Plant Loop Volume [m3] | 1.299 | 0.649 | 0.433 | 0.325 | 0.216 | 0.13 | 0.087 | 0.065 | 0.043 | 0.022 |

In PlantManager, when Plant Loop Volume is autosized:

```
PlantLoop(LoopNum)%Volume = PlantLoop(LoopNum)%MaxVolFlowRate*TimeStepZone*SecInHour/0.8d0
```

which would be the Plant Loop Volume during a timestep.  Instead, shouldn't this be:

```
PlantLoop(LoopNum)%Volume = PlantLoop(LoopNum)%MaxVolFlowRate*SecInHour/0.8d0
```

As far as I can tell, PlantLoop(LoopNum)%Volume is only used to calculate the PlantLoop(LoopNum)%Mass a few lines down in PlantManger:

```
PlantLoop(LoopNum)%Mass = PlantLoop(LoopNum)%Volume * FluidDensity
```

Then later in HVACInterfaceManager.f90:

   ThisTankMass = FracTotLoopMass \* PlantLoop(LoopNum)%Mass

*MJW 06 Feb 2017 - Added defect files in EnergyPlusDevSupport to test time-step dependency and the new input field for both PlantLoop and CondenserLoop obejcts.*

**Comment by mjwitte** [comment]
@EnergyArchmage The autocalculation of plant loop volume is still timestep dependent as noted in this issue.  Changing this will cause broad diffs, because nearly every file autocalculates the plantloop volume.  Current code is:

`PlantLoop(LoopNum)%Volume = PlantLoop(LoopNum)%MaxVolFlowRate*TimeStepZone*SecInHour/0.8d0`

With the default timestep of 6, TimeStepZone=1/6, so suggest changing to 

`PlantLoop(LoopNum)%Volume = PlantLoop(LoopNum)%MaxVolFlowRate*(1/6)*SecInHour/0.8d0`
or
`PlantLoop(LoopNum)%Volume = PlantLoop(LoopNum)%MaxVolFlowRate*SecInHour/4.8d0`

This should result in no diffs for any files using Timestep=6 (or default timestep).  Would you agree with this change?

**Comment by rglazer2** [comment]
@mjwitte I could be wrong about this, but doesn't it make sense to completely remove the "TimeStepZone" part of the PlantLoop.Volume calculation, as it seems to imply that the loop volume is different depending on the timestep.

I did some searching and it seems like the only place using PlantLoop.Volume is the PlantLoop.Mass calculation just below in the same file.

PlantLoop.Mass is apparently only used in HVACInterfaceManager.cc to calculate "ThisTankMass," which doesn't seem to take into account timestep length, so this calculation is probably carrying over the same error as PlantLoop Volume.

**Comment by mjwitte** [comment]
@rglazer2 The improvement in what I proposed is that we would use the same constant regardless of the value of TimeStepZone.  The suggestion to use SecInHour/4.8 was just to keep most simulations having the same result.  

I would like input from @EnergyArchmage - if you were to start from scratch to come up with a default ratio for plant volume to plant max flow rate, what would you choose?

**Comment by EnergyArchmage** [comment]
I agree that the current method of autosizing plant loop volume is suspect and in the past I have also wanted to change it.  The problem is that I don't know what to change it to.  There would need to be some kind of study of a sample of real buildings to find a correlation as a function of max flow rate ( and get some pump head data at the same time).   

Besides this issue, I think it produces too large of a result.  For example the 68.9 m3 result above for CW plant at 6 timesteps is 18,500 gallons of water, which seems like a lot to me.  So if this does get changed, I would favor also changing it to yield smaller values.  So if forced, I might choose 1/10 instead of 1/6 for Mike's suggestion.  

The current method was developed before my time, but I think it was just a trial and error sort of thing with the criteria to yield a value that stabalizes the numerical model.  So the timestep (probably) got in there for numercial reasons.  (The damping of a bigger volume helps keep temperatures from spiking over a long timestep. )  However plant solver has since changed and now can even handle very tiny volumes (if hardsized), so I think it will be safe to get timestep out of the correlation.

**Comment by mjwitte** [comment]
I contacted a designer for a rule of thumb:

>  Often the manufacturers specify some minimum volume based on the chiller capacity, so a rule of thumb number might be 4 gallons per ton of chiller capacity although this could vary among vendors.  This volume could be related to the flowrate of the chiller if you assumed 2.4 gpm per ton based on 10F temperature difference or 2 gpm based on 12F DT…this wouldn’t be the same in all plants, it might be common to see DT range from from 8F to 15F, with the most common at either 10F or 12F. (Not usually a lot higher than 15F, that would be rare.)

If we start with 4 gal/ton \* (ton/2 gpm) = 4/2 gal/gpm = 2 gal/gpm  which means the loop turns over every 2 minutes, give or take.

Volume = 2 min \* flow rate

In SI Volume [m3] = Flow [ m3/s] \* 2 min \* 60 s/min = Flow \* 120 compared to the 3600/4.8=750 that I suggested, or the 450 that Brent suggested.

**Comment by mjwitte** [comment]
Found this document online which support the 2 minute factor.

On p. 2 of [Chilled Water System Design Issues](https://www.trane.com/content/dam/Trane/Commercial/global/products-systems/education-training/engineers-newsletters/waterside-design/ADM-APN051-EN.pdf)

> In this case, the required loop time
> was two minutes. So the system water
> volume required is the loop time
> multiplied by the water flow:
> 2 min. x 300 gpm = 600 gal

**Comment by mjwitte** [comment]
@rglazer2 Let's try this with a 2 minute factor and see what kind of diffs we get.  We'll need to find where (if it's there at all) this is dicsussed in the Engineering Reference.  And we could also add a line in the I/O Reference Plantloop Volume field description indicating what the autocalculate will do.

```
// Autocalculated loop volume based on a loop time of 2 minutes
PlantLoop(LoopNum)%Volume = PlantLoop(LoopNum)%MaxVolFlowRate*120.0d0
```

It would also be good to check user input values for Volume and throw a warning if it's too large. If Volume/MaxVolFlowRate > 3600 (1 hour) then warn something like "PlantLoop "<loop name>" Plant Loop Volume is high relative to the Maximum Loop Flow Rate. The loop time is nn hr."

**Comment by mjwitte** [comment]
@rglazer2 is working on this.

**Comment by mjwitte** [comment]
@rglazer2 Would be nice to address #4126 along with this fix.

**Comment by Myoldmopar** [comment]
Closed via #6000
